stl/inc/list

```diff
@@ -745,8 +745,8 @@ struct _List_node_insert_op2 {
 private:
     _Alnode& _Al;
     size_type _Added; // if 0, the values of _Head and _Tail are indeterminate
-    pointer _Tail; // points to the most recently appended element; it doesn't have _Next constructed
-    pointer _Head; // points to the first appended element; it doesn't have _Prev constructed
+    pointer _Tail{}; // points to the most recently appended element; it doesn't have _Next constructed
+    pointer _Head{}; // points to the first appended element; it doesn't have _Prev constructed

```

Comment:
User-provided types again; I think `= nullptr` (or even `= pointer()`) would be an improvement here.

Reply:
Per Discord discussion, this is fairly conventional for fancy pointers. We should fix all such occurrences (in a different PR) or none.

Reply:
\[nullablepointer.requirements\] says Cpp17NullablePointer requires Cpp17DefaultConstructible, and \[tab:cpp17.defaultconstructible\] says that `T{}` must be either value-initialized or aggregate-initialized. So this is valid, and we routinely do this in `forward_list` and `list` for potentially fancy pointers.

Reply:
...and Stephan points out that _`Cpp17NullablePointer`_ requires _`Cpp17DefaultConstructible`_ which requires `T{}` to be valid, so I'm simply wrong.